universal-application-tool-0.0.1/app/repository/ApplicantRepository.java

```diff
@@ -45,23 +52,29 @@ public ApplicantRepository(EbeanConfig ebeanConfig, DatabaseExecutionContext exe
    */
   public CompletionStage<ImmutableList<ProgramDefinition>> programsForApplicant(long applicantId) {
     return supplyAsync(
-            () ->
-                ebeanServer
-                    .find(Program.class)
-                    .alias("p")
-                    .where()
-                    .or()
-                    .eq("lifecycle_stage", LifecycleStage.ACTIVE)
-                    .exists(
-                        ebeanServer
-                            .find(Application.class)
-                            .where()
-                            .eq("applicant.id", applicantId)
-                            .eq("lifecycle_stage", LifecycleStage.DRAFT)
-                            .raw("program.id = p.id")
-                            .query())
-                    .endOr()
-                    .findList(),
+            () -> {
+              List<Program> inProgressPrograms =
+                  ebeanServer
+                      .find(Program.class)
+                      .alias("p")
+                      .where()
+                      .exists(
+                          ebeanServer
+                              .find(Application.class)
+                              .where()
+                              .eq("applicant.id", applicantId)
+                              .eq("lifecycle_stage", LifecycleStage.DRAFT)
```

Comment:
Question (non-blocking): looks like applications use the same lifecycle stages but they aren't associated with versions?

Reply:
Correct, yeah, we already have it so we might as well, I think.